freud/common.pyx

```diff
@@ -125,6 +133,70 @@ cdef class Compute:
         return wrapper
 
 
+cdef class PairCompute(Compute):
+    R"""Parent class for all compute classes in freud that depend on finding
+    nearest neighbors.
+
+    The purpose of this class is to consolidate some of the logic for parsing
+    the numerous possible inputs to the compute calls of such classes. In
+    particular, this class contains a helper function that calls the necessary
+    functions to create NeighborQuery and NeighborList classes as needed, as
+    well as dealing with boxes and query arguments.
+
+    .. moduleauthor:: Vyas Ramasubramani <vramasub@umich.edu>
+    """
+    def preprocess_arguments(self, box, points, query_points=None, nlist=None,
+                              query_args=None):
+        """Process standard compute arguments into freud's internal types by
+        calling all the required internal functions.
+
+        The preprocessing of boxes and points into NeighborQuery objects, the
+        determination of how to handle the NeighborList object, the creation
+        of default query arguments as needed, deciding what query_points are,
+        and setting the appropriate exclude_ii flag are all handled in this
+        function.
+
+        Args:
+            box (:class:`freud.box.Box`):
+                Simulation box.
+            points ((:math:`N_{points}`, 3) :class:`numpy.ndarray`):
+                Reference points used to calculate the RDF.
+            query_points ((:math:`N_{query_points}`, 3) :class:`numpy.ndarray`, optional):
+                Points used to calculate the RDF. Uses :code:`points` if
+                not provided or :code:`None`.
+            nlist (:class:`freud.locality.NeighborList`, optional):
+                NeighborList to use to find bonds (Default value =
+                :code:`None`).
+            query_args (dict): A dictionary of query arguments (Default value =
+                :code:`None`).
+        """
+        cdef freud.box.Box b = freud.common.convert_box(box)
+
+        cdef freud.locality.NeighborQuery nq = freud.locality.make_default_nq(
+            box, points)
+        cdef freud.locality.NlistptrWrapper nlistptr = \
+            freud.locality.NlistptrWrapper(nlist)
+
+        cdef freud.locality._QueryArgs qargs = \
+            freud.locality._QueryArgs.from_dict(
+                query_args if query_args else
+                self.get_default_query_args(query_points is None))
+        if query_points is None:
+            query_points = nq.points
+        query_points = freud.common.convert_array(
+            query_points, shape=(None, 3))
+        cdef const float[:, ::1] l_query_points = query_points
+        cdef unsigned int n_p = l_query_points.shape[0]
+        return (b, nq, nlistptr, qargs, l_query_points, n_p)
+
+    def get_default_query_args(self, exclude_ii):
```

Comment:
Does it make sense to have this `exclude_ii` parameter, or should it be left undefined in `get_default_query_args` and then just updated by the `preprocess_arguments` method? As far as I can remember, we always choose `exclude_ii` based on whether there is a second set of points provided, so I don't think this option is needed.

Reply:
I was debating about this. I originally had this as a property, then made it a function that depended on `exclude_ii` because I didn't want developers to have to update it everywhere in the compute method. Now that I've moved this logic into the common module, I can change it back to a property and just update the `exclude_ii` logic afterwards.